PyPDF2/pdf.py

```diff
@@ -1195,6 +1200,11 @@ def _showwarning(message, category, filename, lineno, file=warndest, line=None):
 
         self._override_encryption = False
 
+        if "/AcroForm" in self.trailer["/Root"]:
+            self.trailer["/Root"]["/AcroForm"].update({
+                NameObject("/NeedAppearances"): BooleanObject(True)
+            })
+
```

Comment:
```suggestion
```

We now have `writer.set_need_appearances_writer()` which is used in `writer.update_page_form_field_values`